Submatrices, determinantes paso a paso

JonnySC-axl opened the pull request:
Ya estan el programa para las determinantes paso a paso, al igual que los headers y el codigo para crear las submatrices

JonnySC-axl committed:
Submatrices, determinantes paso a paso

P0nchyz commented on determinantesPaP.c:
```diff
@@ -0,0 +1,77 @@
+    // Caso base 2x2
+    if (A->n == 2) {
+        printf("Caso base 2x2. Matriz:\n");
+        mostrarMatriz(A);
```
¿Donde esta definida esa función?

P0nchyz commented on lib/subMatriz.h:
```diff
@@ -0,0 +1,53 @@
+#include "matrices.h"
+
+
+typedef struct SubMatriz {
```
¿Por qué hay una nueva estructura para una submatriz si es igual que el de matriz normal?

P0nchyz commented on lib/subMatriz.h:
```diff
@@ -0,0 +1,53 @@
+/// @param ancho Numero de Columnas
+/// @param alto Numero de Filas
+/// @return Apuntador a la Matriz Creada
+Matriz *crearMatriz(unsigned int ancho, unsigned int alto);
```
Función crearMatriz ya esta definida.

P0nchyz commented on determinantesPaP.c:
```diff
@@ -0,0 +1,77 @@
+    int n = A->n;
+
+    // Submatriz de tamaño (n-1) x (n-1)
+    Matriz sub;
```
Ya hay una función que crea matrices

P0nchyz reviewed (changes requested):
- La función de subMatrices debe de usar el struct matriz , ya que no hay ninguna diferencia con el struct subMatriz.
-  Hay funciones que son declaradas en otras partes del programa que vuelven a ser declaradas en subMatriz.h
- funcion mostrarMatriz no existe

JonnySC-axl committed:
Se hicieron los cambios que pediste

ya se encuentra la función mostrarMatriz que antes no estaba, se cambio el struct de submatriz que ya estaba ( // Submatriz de tamaño (n-1) x (n-1))
a  Matriz *sub = crearMatriz(n - 1, n - 1);
como lo habias pedido

En matriz.h una es *crearmatriz
y la otra es crearmatrizidentidad

JonnySC-axl committed:
SubMatriz a Matriz

Se hicieron los cambios de subMatriz a solo Matriz en el programa de determinantes paso a paso ademas de que se eliminaron las librerias de subMatriz

JonnySC-axl committed:
Submatriz a Matriz

Se elimino el include de la libreria submatriz
y unas cosas del void

JonnySC-axl committed:
Void calldeterminatesPaP

Se cambio el void (calldeterminantePaP) en el main para utilizar matriz en lugar de submatriz
se borro mostrarMatriz

JonnySC-axl committed:
Menu en DPaP

Se volvio a agregar el menu de determinantes paso a paso en el programa de determinantesPaP.c
